components/workspace/Panel.tsx

```diff
@@ -272,79 +276,126 @@ export function WorkspacePanel(props: PropsOf<typeof ActionPanel>) {
-          <>
-            <CheckCircleIcon
-              title="Empty workspace"
-              className="h-10 w-10 text-green-300 align-text-bottom mx-auto mb-4"
-            />
-            <p className="pb-4 text-center text-gray-400 dark:text-gray-50">
-              Workspace is empty.
-            </p>
-            <WorkspaceItemCreator />
-          </>
-        </div>
-      ) : (
-        <>
-          {showItemCreator && (
-            <WorkspaceItemCreator
-              size="sm"
-              onCancel={() => setShowItemCreator(false)}
-            />
-          )}
-          <form ref={formRef} id={WORKSPACE_FORM_ID} onSubmit={onFormSubmit}>
-            {showActionForm && (
-              <WorkspacePanelActionForm
-                {...{
-                  modEventType,
-                  setModEventType,
-                  onCancel: () => setShowActionForm((current) => !current),
-                }}
-              />
-            )}
-            {!showItemCreator && (
-              <div className="mb-2 flex space-x-2">
-                <WorkspacePanelActions
-                  listData={workspaceListStatuses || {}}
-                  handleRemoveSelected={handleRemoveSelected}
-                  handleEmptyWorkspace={handleEmptyWorkspace}
-                  handleFindCorrelation={handleFindCorrelation}
-                  setShowActionForm={setShowActionForm}
-                  setShowItemCreator={setShowItemCreator}
-                  showActionForm={showActionForm}
-                  workspaceList={workspaceList}
-                />
-              </div>
+      <Dropzone
+        accept={{ 'application/json': ['.json'], 'text/csv': ['.csv'] }}
+        onDrop={importFromFiles}
+        ref={dropzoneRef}
+        onDropRejected={(rejections) => {
+          toast.error(
+            rejections
+              .map((r) => r.errors.map((e) => e.message).join(' | '))
+              .flat()
+              .join(' | '),
+          )
+        }}
+        noKeyboard
+        noClick
+      >
+        {({ getRootProps, getInputProps }) => (
+          <div
+            {...getRootProps()}
+            className={
+              !workspaceList?.length
+                ? 'flex flex-col flex-1 h-full item-center justify-center'
+                : ''
+            }
+          >
+            <input {...getInputProps()} />
+            {!workspaceList?.length ? (
+              <>
+                <>
+                  <CheckCircleIcon
+                    title="Empty workspace"
+                    className="h-10 w-10 text-green-300 align-text-bottom mx-auto mb-4"
+                  />
+                  <p className="pb-4 text-center text-gray-400 dark:text-gray-50">
+                    Workspace is empty.
+                  </p>
+                  <WorkspaceItemCreator
+                    onFileUploadClick={() => {
+                      dropzoneRef.current?.open()
+                    }}
+                  />
```

Comment:
This uses two different patterns:
- Data binding through props
- Data binding through (some sort) of context
Sadly I don't see an easy way to do "clean" this. If you do, I'd change this, otherwise let's leave it.